move/axelar_gateway/sources/gateway.move

```diff
@@ -139,14 +139,20 @@ public fun setup(
 // ------
 // Macros
 // ------
-macro fun value($self: &Gateway): &GatewayV0 {
+/// This macro also uses version control to sinplify things a bit.
+macro fun value($self: &Gateway, $function_name: vector<u8>): &GatewayV0 {
     let gateway = $self;
-    gateway.inner.load_value<GatewayV0>()
+    let value = gateway.inner.load_value<GatewayV0>();
+    value.version_control().check(VERSION, $function_name);
```

Comment:
since mysten team might add string literal support in the future, to make this future compatible change version control interface to accept a string. you can convert vector<u8> to string in here for convenience. And for the version control init, you can use a map to convert to strings perhaps